src/lib/ArrayInfo.ts

```diff
@@ -1,121 +1,179 @@
-import {DataError} from "./Errors"
-import {KeyValue} from "./Utils"
+import { DataError } from './Errors'
+import { KeyValue } from './Utils'
 
 function isInt(value: any) {
-    return !isNaN(value) &&
-        Number(value) == value && !isNaN(parseInt(value, 10))
+  return !isNaN(value) && Number(value) == value && !isNaN(parseInt(value, 10))
 }
 
-export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)\[((?!(\]|\[)).*|)\]$/gm
+export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)(.*)/gm
+export const arrayIndiciesRegex = /\[(.*?)\]/g
+export const malformedArrayRegex = /\]|\[/g
```

Comment:
Doesn't need to be an index, can be a simple [include check](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes)

Reply:
This has been resolved in the latest changes.